lib/trino-filesystem/src/main/java/io/trino/filesystem/tracing/TracingInputFile.java

```diff
@@ -85,10 +86,17 @@ public long length()
     public Instant lastModified()
             throws IOException
     {
+        // skip tracing if lastModified is cached, but delegate anyway
```

Comment:
Separate commit?

Why do we need it?

Reply:
It's needed to avoid unnecessary tracing of lastModified due to DefaultCacheKeyProvider in tests even though every TrinoInputFile implementation caches it.

Reply:
I've moved it into separate commit